bot/cogs/last_fm.py

```diff
@@ -0,0 +1,206 @@
+      name="topsongs",
+      brief="Gets a list of the top songs on the world charts",
+      aliases=['songs', 'tops']
+    )
+    async def topsongs(self, ctx):
+        data = top_tracks()
+        embed = discord.Embed(title='Top 10 Tracks', url='https://www.last.fm/charts')
+        for count, song in enumerate(data[:10], 1):
+                embed.add_field(name=count, value=f"{song['name']} by {song['artist']['name']}", inline=False)
+        await ctx.send(embed=embed)
+
+    @commands.command(
+      name="topartists",
+      brief="Gets a list of the top artists on the world charts",
+      aliases=['artists', 'topa']
+    )
+    async def topartists(self, ctx):
+        data = top_artists()
+        embed = discord.Embed(title='Top 10 Artists', url='https://www.last.fm/charts')
+        for count, artist in enumerate(data[:10], 1):
+                embed.add_field(name=count, value=artist['name'], inline=False)
+        await ctx.send(embed=embed)
+
+
+def setup(bot):
+    "Imports the cog"
+    bot.add_cog(MusicCog(bot))
+
+
+def get_data(method, method2=''):
+    """returns the json data from the url, takes in the two pieces of a URL as outlined in the last.fm api docs"""
+    return json.loads(urllib.request.urlopen("http://ws.audioscrobbler.com/2.0/?method="+method+"&api_key="+MUSIC_TOKEN+"&format=json"+method2).read().decode())
+
+
+def get_album(album, artist=''):
+        """returns the json data for a given album, takes in just an album or album and artist"""
+        if not artist:
+                album = urllib.parse.quote(album)
+                data = get_data('album.search&album='+album)
+                album = data['results']['albummatches']['album'][0]['name']
+                artist = data['results']['albummatches']['album'][0]['artist']
+                return get_album(album, artist)
+        else:
+                artist = urllib.parse.quote(artist)
+                album = urllib.parse.quote(album)
+                return get_data('album.getinfo', '&artist='+artist+'&album='+album)
+
+
+def get_artist(artist):
+    """returns the json data for a given artist, takes in artist name"""
+
+    artist = urllib.parse.quote(artist)
+    return get_data('artist.getinfo&artist='+artist)['artist']
+
+
+def get_track(track, artist=''):
+        """Returns info about  a specific song/track takes in just a track or artist and track."""
+        artist = urllib.parse.quote(artist)
+        track = urllib.parse.quote(track)
+
+        if artist:
+        	data = get_data('track.getinfo', '&artist='+artist+'&track='+track)['track']
+            return data
+        else:
+        	artist = get_data('track.search&track='+track)['results']['trackmatches']['track'][0]['artist']
+          	track = get_data('track.search&track='+track)['results']['trackmatches']['track'][0]['name']
+     	    return get_track(track, artist)
+
+
+def top_tracks():
+        """Returns data of the top songs on the charts, requires no args."""
```

Comment:
No need to explicitly say `requires no args` this is very clear from there being no arguments in the function. Otherwise decent docstring, maybe include the number of songs returned.